lib/services/settings/mobile_settings_service.dart

```diff
@@ -58,16 +60,24 @@ class MobileSettingsService extends SettingsService {
   }
 
   @override
-  bool get themeDarkMode {
-    var theme = _sharedPreferences.getString('theme') ?? 'dark';
-
-    return theme == 'dark';
+  String get themeMode {
+    var value = _sharedPreferences.getString('theme') ?? ThemeMode.system.name;
+    if (value == ThemeMode.system.name) {
+      var brightness = SchedulerBinding.instance.platformDispatcher
+          .platformBrightness;
+      return brightness == Brightness.dark ? ThemeMode.dark.name : ThemeMode.light.name;
```

Comment:
But here, you effectively overwrite system with dark or light depending upon the current system theme. Whilst you have changed the return type from bool to String, you are effectively still returning a boolean of either light or dark. This works fine for setting the overall theme, but breaks the setting page.